nvfuser/pytorch_utils.py

```diff
@@ -173,9 +173,6 @@ def clear_cuda_cache() -> None:
     """
     Utility function to clear CUDA cache before running a test.
     """
-    if (
-        torch.cuda.memory_allocated()
```

Comment:
Can we safely remove this? 
Shouldn't we always clear the L2 cache if there is memory allocated at the beginning to benchmark each round from the same memory state?

I was under the impression that this just queries the device statistics, how does this use malloc? Can you point me to any reference on its working?

Reply:
As mentioned [here](https://pytorch.org/docs/stable/notes/cuda.html#cuda-memory-management), "allocated" refers to only the memory occupied by tensors currently, but deleted tensors' memory may still remain allocated to the pytorch cache. That unused but still allocated memory is included along with the allocated memory in the "reserved" amount. I don't think we care what is allocated, and in particular having this here means that we will clear the cache _every_ time this is called if there is even a single tensor still in memory somewhere i.e. not yet garbage collected.